src/site/stages/build/plugins/add-asset-hashes.js

```diff
@@ -1,69 +1,85 @@
+  const teamsiteInjectionBundles = {
+    'proxy-rewrite.js': 'generated/proxy-rewrite.entry.js',
+    'vendor.js': 'generated/vendor.entry.js',
+    'polyfills.js': 'generated/polyfills.entry.js',
+    'styleConsolidated.css': 'generated/styleConsolidated.css',
+    'static-pages.css': 'generated/static-pages.css',
+  };
 
-    // The manifest contains the original filenames without the addition of .entry
-    // on the JS files. This finds all of those and modifies them to add .entry.
-    const manifest = {};
-    Object.keys(originalManifest).forEach(originalManifestKey => {
-      const matchData = originalManifestKey.match(/(.*)\.js$/);
-      if (matchData !== null) {
-        const newKey = `${matchData[1]}.entry.js`;
-        manifest[newKey] = originalManifest[originalManifestKey];
+  for (const fileManifestKey of Object.keys(teamsiteInjectionBundles)) {
+    const teamSitePath = teamsiteInjectionBundles[fileManifestKey];
+    const hashedFileName = entryNamesDictionary.get(fileManifestKey);
+
+    if (!hashedFileName) {
+      if (buildOptions.entry) {
+        // A command arg was passed to build only certain areas of the site, so
+        // it makes that it would be missing.
+        continue;
       } else {
-        manifest[originalManifestKey] = originalManifest[originalManifestKey];
-      }
-    });
-
-    // For each file in the build, if it is a HTML or CSS file, loop over all
-    // the keys in the manifest object and do a search and replace for the
-    // key with the value.
-    Object.keys(files).forEach(filename => {
-      if (filename.match(/\.(html|css)$/) !== null) {
-        Object.keys(manifest).forEach(originalAssetFilename => {
-          const newAssetFilename = manifest[originalAssetFilename].replace(
-            '/generated/',
-            '',
-          );
-          const file = files[filename];
-          const contents = file.contents.toString();
-          const regex = new RegExp(originalAssetFilename, 'g');
-          file.contents = new Buffer(contents.replace(regex, newAssetFilename));
-        });
+        throw new Error('Missing Webpack assets required for TeamSite!');
       }
-    });
-
-    if (!isPreview) {
-      // Create a copy of the proxy-write files without cache-bust hashes
-      [
-        'proxy-rewrite.entry.js',
-        'styleConsolidated.css',
-        'static-pages.css',
-        'vendor.entry.js',
-        'polyfills.entry.js',
-      ].forEach(unhashedName => {
-        const hashedName = manifest[unhashedName];
-
-        // When an --entry is specified that isn't proxy-rewrite, these files won't be here
-        if (hashedName) {
-          files[`generated/${unhashedName}`] = files[hashedName.substr(1)]; // eslint-disable-line no-param-reassign
-        }
+    }
+
+    const hashedFileNameWithoutLeadingSlash = hashedFileName.slice(1);
+    const file = files[hashedFileNameWithoutLeadingSlash];
+
+    if (!file) continue;
+
+    files[teamSitePath] = file;
+  }
+}
```

Comment:
I hate this TeamSite stuff, but it's necessary. View the source for https://benefits.va.gov/ to see the hardcoded references to these assets.